common/web/lm-worker/src/main/model-compositor.ts

```diff
@@ -651,19 +651,21 @@ export default class ModelCompositor {
     return reversion;
   }
 
-  applyReversion(reversion: Reversion, context: Context): Suggestion[] {
+  async applyReversion(reversion: Reversion, context: Context): Promise<Suggestion[]> {
     // If we are unable to track context (because the model does not support LexiconTraversal),
     // we need a "fallback" strategy.
     let compositor = this;
     let fallbackSuggestions = function() {
       let revertedContext = models.applyTransform(reversion.transform, context);
-      let suggestions = compositor.predict({insert: '', deleteLeft: 0}, revertedContext);
-      suggestions.forEach(function(suggestion) {
-        // A reversion's transform ID is the additive inverse of its original suggestion;
-        // we revert to the state of said original suggestion.
-        suggestion.transformId = -reversion.transformId;
+      return compositor.predict({insert: '', deleteLeft: 0}, revertedContext).then((suggestions) => {
+        suggestions.forEach(function(suggestion) {
+          // A reversion's transform ID is the additive inverse of its original suggestion;
+          // we revert to the state of said original suggestion.
+          suggestion.transformId = -reversion.transformId;
+        });
+
+        return suggestions;
       });
```

Comment:
Can we use `async`/`await` pattern rather than mixing `async` and `.then`?
```suggestion
      const suggestions = await compositor.predict({insert: '', deleteLeft: 0}, revertedContext);
      suggestions.forEach(function(suggestion) {
        // A reversion's transform ID is the additive inverse of its original suggestion;
        // we revert to the state of said original suggestion.
        suggestion.transformId = -reversion.transformId;
      });

      return suggestions;
```